src/main/webapp/js/user-map-loader.js

```diff
@@ -11,9 +11,75 @@ function createMap(){
   //creates listener for clicks
   map.addListener('click', (event) => {
     //adds marker wherever user clicks
-    new google.maps.Marker({
-      position: event.latLng,
-      map: map
-    });  
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+}
+
+//creates marker at given lat and lng coordinates with 
+  //corresponding info window
+function createMarkerForEdit(lat, lng){
+  //closes previously opened marker if exists
+  if(editMarker){
```

Comment:
Saw this chunk a few times, might it be a good idea to separate into a helper function?

Reply:
which part?

Reply:
The part where you are deleting the marker, its right below this, starting with if(editMarker) { ... }

Reply:
setMap is the only line that deletes the marker I think and it's already a separate function